src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -218,14 +218,15 @@ def get_current_account_user(self):
 
         return active_account[_USER_ENTITY][_USER_NAME]
 
-    def get_subscription(self, subscription_id=None):
+    def get_subscription(self, subscription_id=None, subscription_name=None):
```

Comment:
Why not simply call this `subscription`? The effect of calling the method and passing the command's `subscription_name_or_id` field into both parameters has the same effect of checking both sets of conditional logic but makes the method signature seem odd. You can simply document in the docstring for the method that `subscription` accepts a name or ID.